File: src/test/java/dev/retrotv/crypt/owe/crc/CRC32Test.java
package dev.retrotv.crypt.owe.crc;

import dev.retrotv.crypt.owe.OWETest;
import dev.retrotv.enums.HashAlgorithm;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class CRC32Test extends OWETest {

    @Test
    @DisplayName("CRC32 File hash 테스트")
    void fileHashTest() throws Exception {
        fileHashTest(HashAlgorithm.CRC32);
    }

    @Test
    @DisplayName("CRC32 File hash matches 테스트")
    void fileHashMatchesTest() throws Exception {
        fileHashMatchesTest(new CRC32(), HashAlgorithm.CRC32);
    }

    @Test
    @DisplayName("CRC32 File and File matches 테스트")
    void fileMatchesTest() throws Exception {
        fileMatchesTest(new CRC32());
    }

    @Test
    @DisplayName("CRC32 password encode 테스트")
    void passwordEncrypt() {
        passwordEncryptAndMatchesTest(new CRC32());
    }
}

File: src/test/kotlin/dev/retrotv/crypto/owe/hash/crc/CRC32Test.kt
package dev.retrotv.crypto.owe.hash.crc

import dev.retrotv.crypto.owe.OWETest
import dev.retrotv.enums.Algorithm
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

internal class CRC32Test : OWETest() {
    @Test
    @DisplayName("CRC32 File hash 테스트")
    @Throws(Exception::class)
    fun fileHashTest() {
        fileHashTest(Algorithm.Hash.CRC32)
    }

    @Test
    @DisplayName("CRC32 File hash matches 테스트")
    @Throws(Exception::class)
    fun fileHashMatchesTest() {
        fileHashMatchesTest(CRC32(), Algorithm.Hash.CRC32)
    }

    @Test
    @DisplayName("CRC32 File and File matches 테스트")
    @Throws(Exception::class)
    fun fileMatchesTest() {
        fileMatchesTest(CRC32())
    }

    @Test
    @DisplayName("CRC32 password encode 테스트")
    fun passwordEncrypt() {
        passwordEncryptAndMatchesTest(CRC32())
    }
}
